dali/test/python/test_fw_iterators.py

```diff
@@ -1051,6 +1052,33 @@ def get_data():
             counter += 1
     assert counter == iter_limit * runs
 
+def check_external_source_variable_size(Iterator, *args, iter_size=-1, **kwargs):
+    batch_size = 4
+    iter_limit = 4
+    runs = 3
+    test_data_shape = [2, 3, 4]
+    i = 0
+    def get_data():
+        nonlocal i
+        if i == iter_limit:
+            i = 0
+            raise StopIteration
+        out = [[np.random.randint(0, 255, size = test_data_shape, dtype = np.uint8) for _ in range(random.randint(1, batch_size))]]
+        i += 1
+        return out
+
+    pipe = Pipeline(batch_size = batch_size, num_threads = 1, device_id = 0)
+    with pipe:
+        outs = fn.external_source(source = get_data, num_outputs=1)
+    pipe.set_outputs(*outs)
+
+    it = Iterator([pipe], *args, auto_reset=True, size=iter_size, **kwargs)
+    counter = 0
+    for _ in range(runs):
+        for __ in enumerate(it):
+            counter += 1
```

Comment:
You can get the data here and compare with `dataset`, so you'll know for sure that data is intact.

Reply:
See the comment above.